### ISSUE TITLE
iso639-2 manager doesn't differenciate between Terminology & Bibliography entries

### ISSUE THREAD

**Issue by geoffroy-noel-ddh** [kingsdigitallab/django-controlled-vocabulary#10]
![image](https://user-images.githubusercontent.com/3778106/88420728-3ba08c80-cddf-11ea-88e9-3c5e502297a5.png)

```
$ pm vocab search -f iso639-2 -p ger
iso639-2: <class 'controlled_vocabulary.vocabularies.iso639_2.VocabularyISO639_2'>
[['deu', 'German'], ['ger', 'German'], ['gmh', 'German, Middle High (ca.1050-1500)'], ['goh', 'German, Old High (ca.750-1050)'], ['gem', 'Germanic languages'], ['lim', 'Limburgan |  Limburger |  Limburgish'], ['nds', 'Low German |  Low Saxon |  German, Low |  Saxon, Low'], ['nic', 'Niger-Kordofanian languages'], ['gsw', 'Swiss German |  Alemannic |  Alsatian']]
done (vocab search)

$ pm vocab search -f iso639-2 -p german
iso639-2: <class 'controlled_vocabulary.vocabularies.iso639_2.VocabularyISO639_2'>
[['ger', 'German'], ['gmh', 'German, Middle High (ca.1050-1500)'], ['goh', 'German, Old High (ca.750-1050)'], ['gem', 'Germanic languages'], ['nds', 'Low German |  Low Saxon |  German, Low |  Saxon, Low'], ['gsw', 'Swiss German |  Alemannic |  Alsatian']]
```

https://www.loc.gov/standards/iso639-2/php/code_list.php

iso369-2 can have two termids for the same language. For instance German has deu and ger codes. Both are valid but they belong to two different categories: ger is categorised as Bibliographic (B), deu as Terminology (T). B is the shorten version of the english name of the language whereas T comes from the native name of the language. Same with fre vs fra for French.

The standard recommends to pick one and remain consistent with that choice. Unfortunately tihs was overlooked in out implementation and we distinguish between those, so there is no guarantee that the first entry is always B type.

Consequences:

1. if we keep both, we should provide information through the API and on screen to help the user tell them apart (see screenshot)
2. the sort order from the API should also be deterministic and consistent (see issue with the two calls above: deu comes first in first lookup but ger comes first in second lookup (and deu disappears!))

**Comment by geoffroy-noel-ddh** [comment]
Suggestions:
1. we keep both versions in the API because a) they might have already been used by a project so we can't break that b) it would be an arbitrary choice to pick one that might fustrate users if they can't find either ger or deu
2. in case of ambiguity in the label, we indicate in the term description whether it is a T or B (so it appears on screen and in the API results). This keeps the label intact, which is good because it is faithful to the standard and this allows us to find 'synonyms' in the databse without relying on naming conventions (i.e. all German terms have the exact 'German' label).
3. we need to fetch a better vocabulary file than the one we use now as it doesn't distinguish between T & B!
4. the API will always return B termids before the T termids. It is alsothe termid that will be returned by utils.search_or_none(). According to the new source file this is the primary termid.

**Comment by geoffroy-noel-ddh** [comment]
Fixed all the issues above. Released in 0.7.

One outstanding issue is that once the term is selected with the widget, it doesn't show T or B so the actual termid behind the selction remains ambiguous.
